cmd/qq/main.go

```diff
@@ -0,0 +1,48 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/hpcloud/tail"
+)
+
+var (
+	logfile string
+	logpath string
+)
+
+func init() {
+	logfile = "q"
+	setPath()
+}
+
+func main() {
+	tailFile()
+}
+
+func tailFile() {
+	t, err := tail.TailFile(logpath, tail.Config{Follow: true})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	for line := range t.Lines {
+		fmt.Println(line.Text)
+	}
+}
+
+func setPath() {
+	tmpdir := os.Getenv("TMPDIR")
+	if tmpdir == "" {
+		if _, err := os.Stat("/system/bin/adb"); os.IsNotExist(err) {
+			// Handle android
+			logpath = "/data/local/tmp/" + logfile
+		} else {
+			logpath = "/tmp/" + logfile
+		}
+	} else {
+		logpath = tmpdir + logfile
+	}
+}
```

Comment:
Here's how I'd rewrite this:
```go
func main() {
    path := filepath.Join(os.TempDir(), "q")
    t, err := tail.TailFile(path, tail.Config{Follow: true})
    if err != nil {
        log.Fatal(err)
    }

    for line := range t.Lines {
        fmt.Println(line.Text)
    }
}
```
I shaved off a few lines and there's no `init()` and no globals :)